exonum/src/blockchain/block.rs

```diff
@@ -56,52 +56,9 @@ pub struct Block {
     pub tx_hash: Hash,
     /// Hash of the blockchain state after applying transactions in the block.
     pub state_hash: Hash,
-}
-
-impl Block {
-    /// Create new `Block`.
-    pub fn new(
-        proposer_id: ValidatorId,
-        height: Height,
-        tx_count: u32,
-        prev_hash: Hash,
-        tx_hash: Hash,
-        state_hash: Hash,
-    ) -> Self {
-        Self {
-            proposer_id,
-            height,
-            tx_count,
-            prev_hash,
-            tx_hash,
-            state_hash,
-        }
-    }
-    /// Identifier of the leader node which has proposed the block.
-    pub fn proposer_id(&self) -> ValidatorId {
-        self.proposer_id
-    }
-    /// Height of the block, which is also the number of this particular
-    /// block in the blockchain.
-    pub fn height(&self) -> Height {
-        self.height
-    }
-    /// Number of transactions in this block.
-    pub fn tx_count(&self) -> u32 {
-        self.tx_count
-    }
-    /// Hash link to the previous block in the blockchain.
-    pub fn prev_hash(&self) -> &Hash {
-        &self.prev_hash
-    }
-    /// Root hash of the Merkle tree of transactions in this block.
-    pub fn tx_hash(&self) -> &Hash {
-        &self.tx_hash
-    }
-    /// Hash of the blockchain state after applying transactions in the block.
-    pub fn state_hash(&self) -> &Hash {
-        &self.state_hash
-    }
+    /// Root hash of the Merkle Patricia tree of the erroneous calls performed within the block.
+    /// These calls can include transactions and/or `before_commit` hooks for services.
+    pub error_hash: Hash,
```

Comment:
curious, can we aggregate `tx_hash` and `error_hash` into the `state_hash` like aggregation to make block header structure more extendable? 
And what are the concerns to add this hashes into `state_hash`?